Added warning and small updates (#927)
openhab/openhab-docs: configuration/jsr223.md

## Overview

[JSR223](https://docs.oracle.com/javase/6/docs/technotes/guides/scripting/) ([spec](https://jcp.org/aboutJava/communityprocess/pr/jsr223/index.html)) is a standard scripting API for Java Virtual Machine (JVM) [languages](https://en.wikipedia.org/wiki/List_of_JVM_languages). 
The JVM languages provide varying levels of support for the JSR223 API and interoperability with the Java runtime. 
Currently the following languages are known to work well for openHAB scripting: 
[**Jython**](https://jython.github.io/) (Python on the JVM), 
[**Nashorn Javascript**](http://www.oracle.com/technetwork/articles/java/jf14-nashorn-2126515.html) (ECMAScript implementation included in JDK8 through 10, [deprecated in 11](https://openjdk.java.net/jeps/335)), and 
[**Apache Groovy**](http://www.groovy-lang.org/) (JVM scripting language).

Although JSR223 scripts can be used as a general-purpose extension language for openHAB, it is most commonly used for the creation of rules, and within scripted Actions or Conditions. 
Currently, openHAB allows JSR223 scripting to access all packages, which may not be included in the official APIs. 

> This provides great flexibility for the users of JSR223, but is also 'use at your own risk', since changes outside of the offical APIs occur frequently, and are not considered to be 'breaking changes'.

Reviewer: ```suggestion
This provides great flexibility for the users of JSR223, but is also *use at your own risk*, since changes outside of the offical APIs occur frequently, and are not considered to be *breaking changes*.
```

New APIs are planned to be implemented in the future, which will provide standardized interfaces for interacting with openHAB through scripted automation.

### Language-Specific Documentation

- [Jython](jsr223-jython.html)
- [Javascript](jsr223-js.html)
- Groovy (TODO)

### Script Locations

> Scripts should be placed in the '${OPENHAB_CONF}/automation/jsr223/' directory. 

Reviewer: ```suggestion
Scripts should be placed in the `${OPENHAB_CONF}/automation/jsr223/` directory. 
```

| Parameter | Description | Type |
|--|--|--|
| `itemName` | The item name | TEXT |
| `command` | The command | TEXT |

**Trigger: `core.ItemStateUpdateTrigger`**

| Parameter | Description | Type |
|--|--|--|
| `itemName` | The item name | TEXT |
| `state` | The state | TEXT |

**Trigger: `core.ItemStateChangeTrigger`**

| Parameter | Description | Type |
|--|--|--|
| `itemName` | The item name | TEXT |
| `previousState` | The previous state | TEXT |
| `state` | The state | TEXT |

**Trigger: `timer.GenericCronTrigger`**

| Parameter | Description | Type |
|--|--|--|
| `cronExpression` | The cron expression | TEXT |

**Trigger: `timer.TimeOfDayTrigger`**

| Parameter | Description | Type |
|--|--|--|
| `time` | The time in "hh:mm" format | TEXT |

**Trigger: `core.GenericEventTrigger`**

| Parameter | Description | Type |
|--|--|--|
| `eventTopic` | default: "smarthome/*" (for channel events: "smarthome/channels/*/triggered") | TEXT |
| `eventSource` | item name, channel UID | TEXT |
| `eventTypes` | ItemCommandEvent, ItemStateEvent | TEXT |

### Additional Information

- [Scripted Rule Support](https://github.com/eclipse/smarthome/wiki/Scripted-Rule-Support) - 

>     Early documentation on JSR223 usage in openHAB 2, incliuding IDE setup instructions

Reviewer: ```suggestion
    Early documentation on JSR223 usage in openHAB 2, including IDE setup instructions
```

<a name="presets"></a>

### Predefined Script Variables (all JSR223 languages)

> To faciliate JSR223 scripting, several openHAB-related variables are automatically predefined, and can be loaded into the script context using `scriptExtension.importPreset(<name of preset>)`.

Reviewer: ```suggestion
To faciliate JSR223 scripting, several openHAB-related variables are automatically predefined and can be loaded into the script context using `scriptExtension.importPreset(<name of preset>)`.
```
